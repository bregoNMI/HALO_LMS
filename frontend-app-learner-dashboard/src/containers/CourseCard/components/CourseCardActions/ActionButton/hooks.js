import { useWindowSize, breakpoints } from '@openedx/paragon';

export const useIsCollapsed = () => {
  const { width } = useWindowSize();
  return width < breakpoints.medium.maxWidth && width > breakpoints.small.maxWidth;
};

export default useIsCollapsed;
