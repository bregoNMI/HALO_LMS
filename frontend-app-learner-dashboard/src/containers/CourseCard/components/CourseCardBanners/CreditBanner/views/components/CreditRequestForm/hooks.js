import React from 'react';

export const useCreditRequestFormData = (requestData) => {
  const ref = React.useRef(null);
  React.useEffect(() => {
    if (requestData !== null) {
      ref.current.click();
    }
  }, [requestData]);
  return { ref };
};

export default useCreditRequestFormData;
