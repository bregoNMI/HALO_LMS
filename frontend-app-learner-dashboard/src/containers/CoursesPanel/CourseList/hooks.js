import { useWindowSize, breakpoints } from '@openedx/paragon';

export const useIsCollapsed = () => {
  const { width } = useWindowSize();
  return width < breakpoints.medium.maxWidth;
};

export default useIsCollapsed;
