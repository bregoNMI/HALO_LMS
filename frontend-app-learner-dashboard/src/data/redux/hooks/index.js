export * from './app';
export * from './requests';
