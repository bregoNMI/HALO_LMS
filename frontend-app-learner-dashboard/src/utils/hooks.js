import { useIntl } from '@edx/frontend-platform/i18n';

import dateFormatter from './dateFormatter';

export const useFormatDate = () => {
  const { formatDate } = useIntl();
  return (date) => dateFormatter(formatDate, date);
};

export default {
  useFormatDate,
};
