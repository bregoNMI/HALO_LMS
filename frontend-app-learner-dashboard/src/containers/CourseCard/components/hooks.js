import { reduxHooks } from 'hooks';

export const useActionDisabledState = (cardId) => {
  const { isMasquerading } = reduxHooks.useMasqueradeData();
  const {
    canUpgrade, hasAccess, isAudit, isAuditAccessExpired,
  } = reduxHooks.useCardEnrollmentData(cardId);
  const {
    isEntitlement, isFulfilled, canChange, hasSessions,
  } = reduxHooks.useCardEntitlementData(cardId);

  const { resumeUrl, homeUrl, upgradeUrl } = reduxHooks.useCardCourseRunData(cardId);

  const disableBeginCourse = !homeUrl || (isMasquerading || !hasAccess || (isAudit && isAuditAccessExpired));
  const disableResumeCourse = !resumeUrl || (isMasquerading || !hasAccess || (isAudit && isAuditAccessExpired));
  const disableViewCourse = !hasAccess || (isAudit && isAuditAccessExpired);
  const disableSelectSession = !isEntitlement || isMasquerading || !hasAccess || (!canChange || !hasSessions);
  const disableUpgradeCourse = !upgradeUrl || (isMasquerading && !canUpgrade);

  const disableCourseTitle = (isEntitlement && !isFulfilled) || disableViewCourse;

  return {
    disableBeginCourse,
    disableResumeCourse,
    disableViewCourse,
    disableUpgradeCourse,
    disableSelectSession,
    disableCourseTitle,
  };
};

export default useActionDisabledState;
