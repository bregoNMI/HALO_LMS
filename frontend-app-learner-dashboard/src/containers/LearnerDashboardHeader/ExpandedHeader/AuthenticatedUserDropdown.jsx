import React from 'react';

import { getConfig } from '@edx/frontend-platform';
import { useIntl } from '@edx/frontend-platform/i18n';
import { AppContext } from '@edx/frontend-platform/react';
import { AvatarButton, Dropdown, Badge } from '@openedx/paragon';

import { reduxHooks } from 'hooks';

import messages from '../messages';

export const AuthenticatedUserDropdown = () => {
  const { formatMessage } = useIntl();
  const { authenticatedUser } = React.useContext(AppContext);
  const dashboard = reduxHooks.useEnterpriseDashboardData();

  return (
    authenticatedUser && (
      <Dropdown className="user-dropdown pr4">
        <Dropdown.Toggle
          as={AvatarButton}
          src={authenticatedUser.profileImage}
          id="user"
          variant="light"
          className="p-4"
        >
          <span data-hj-suppress className="d-md-inline">
            {authenticatedUser.username}
          </span>
        </Dropdown.Toggle>
        <Dropdown.Menu className="dropdown-menu-right">
          { getConfig().ENABLE_EDX_PERSONAL_DASHBOARD && (
            <>
              <Dropdown.Header>{formatMessage(messages.dashboardSwitch)}</Dropdown.Header>
              <Dropdown.Item as="a" href="/edx-dashboard" className="active">
                {formatMessage(messages.dashboardPersonal)}
              </Dropdown.Item>
              {!!dashboard && (
                <Dropdown.Item as="a" href={dashboard.url} key={dashboard.label}>
                  {dashboard.label} {formatMessage(messages.dashboard)}
                </Dropdown.Item>
              )}
              <Dropdown.Divider />
            </>
          )}

          {!dashboard && getConfig().CAREER_LINK_URL && (
            <Dropdown.Item href={`${getConfig().CAREER_LINK_URL}`}>
              {formatMessage(messages.career)}
              <Badge className="px-2 mx-2" variant="warning">
                {formatMessage(messages.newAlert)}
              </Badge>
            </Dropdown.Item>
          )}
          <Dropdown.Item href={`${getConfig().ACCOUNT_PROFILE_URL}/u/${authenticatedUser.username}`}>
            {formatMessage(messages.profile)}
          </Dropdown.Item>
          <Dropdown.Item href={getConfig().ACCOUNT_SETTINGS_URL}>
            {formatMessage(messages.account)}
          </Dropdown.Item>
          {getConfig().ORDER_HISTORY_URL && (
            <Dropdown.Item href={getConfig().ORDER_HISTORY_URL}>
              {formatMessage(messages.orderHistory)}
            </Dropdown.Item>
          )}
          <Dropdown.Divider />
          <Dropdown.Item href={getConfig().LOGOUT_URL}>
            {formatMessage(messages.signOut)}
          </Dropdown.Item>
        </Dropdown.Menu>
      </Dropdown>
    )
  );
};

AuthenticatedUserDropdown.propTypes = {};

export default AuthenticatedUserDropdown;
