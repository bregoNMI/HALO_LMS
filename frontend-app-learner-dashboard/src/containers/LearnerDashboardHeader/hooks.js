import React from 'react';
import { useWindowSize, breakpoints } from '@openedx/paragon';
import track from 'tracking';
import { StrictDict } from 'utils';
import { linkNames } from 'tracking/constants';

import * as module from './hooks';

export const state = StrictDict({
  isOpen: (val) => React.useState(val), // eslint-disable-line
});

export const useIsCollapsed = () => {
  const { width } = useWindowSize();
  const isCollapsed = React.useMemo(() => (width <= breakpoints.large.minWidth), [width]);
  return isCollapsed;
};

export const findCoursesNavClicked = (href) => track.findCourses.findCoursesClicked(href, {
  linkName: linkNames.learnerHomeNavExplore,
});

export const findCoursesNavDropdownClicked = (href) => track.findCourses.findCoursesClicked(href, {
  linkName: linkNames.learnerHomeNavDropdownExplore,
});

export const useLearnerDashboardHeaderData = () => {
  const [isOpen, setIsOpen] = module.state.isOpen(false);
  const toggleIsOpen = () => setIsOpen(!isOpen);

  return {
    isOpen,
    toggleIsOpen,
  };
};

export default {
  useIsCollapsed,
  findCoursesNavClicked,
  findCoursesNavDropdownClicked,
  useLearnerDashboardHeaderData,
};
