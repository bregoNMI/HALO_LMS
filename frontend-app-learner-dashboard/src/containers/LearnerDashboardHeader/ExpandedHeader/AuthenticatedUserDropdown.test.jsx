import { shallow } from '@edx/react-unit-test-utils';

import { reduxHooks } from 'hooks';
import { getConfig } from '@edx/frontend-platform';
import { AppContext } from '@edx/frontend-platform/react';
import { AuthenticatedUserDropdown } from './AuthenticatedUserDropdown';
import { useIsCollapsed } from '../hooks';

jest.mock('@edx/frontend-platform', () => ({
  getConfig: jest.fn(),
}));

jest.mock('@edx/frontend-platform/react', () => ({
  AppContext: {
    authenticatedUser: {
      profileImage: 'profileImage',
      username: 'username',
    },
  },
}));
const COURSE_SEARCH_URL = 'test-course-search-url';

jest.mock('hooks', () => ({
  reduxHooks: {
    useEnterpriseDashboardData: jest.fn(),
    usePlatformSettingsData: jest.fn(() => ({
      courseSearchUrl: COURSE_SEARCH_URL,
    })),
  },
}));
jest.mock('../hooks', () => ({
  useIsCollapsed: jest.fn(),
  findCoursesNavDropdownClicked: (href) => jest.fn().mockName(`findCoursesNavDropdownClicked('${href}')`),
}));

jest.mock('data/services/lms/urls', () => ({
  baseAppUrl: (url) => (url),
  programsUrl: 'http://localhost:18000/dashboard/programs',
}));

const config = {
  ACCOUNT_PROFILE_URL: 'http://account-profile-url.test',
  ACCOUNT_SETTINGS_URL: 'http://account-settings-url.test',
  LOGOUT_URL: 'http://logout-url.test',
  ORDER_HISTORY_URL: 'http://order-history-url.test',
  SUPPORT_URL: 'http://localhost:18000/support',
  CAREER_LINK_URL: 'http://localhost:18000/career',
  LMS_BASE_URL: 'http:/localhost:18000',
  ENABLE_EDX_PERSONAL_DASHBOARD: true,
};
getConfig.mockReturnValue(config);

describe('AuthenticatedUserDropdown', () => {
  const defaultDashboardData = {
    label: 'label',
    url: 'url',
  };

  describe('snapshots', () => {
    test('no auth render empty', () => {
      const { authenticatedUser } = AppContext;
      AppContext.authenticatedUser = null;
      const wrapper = shallow(<AuthenticatedUserDropdown />);
      expect(wrapper.snapshot).toMatchSnapshot();
      expect(wrapper.isEmptyRender()).toBe(true);
      AppContext.authenticatedUser = authenticatedUser;
    });
    test('with enterprise dashboard', () => {
      reduxHooks.useEnterpriseDashboardData.mockReturnValueOnce(defaultDashboardData);
      useIsCollapsed.mockReturnValueOnce(true);
      const wrapper = shallow(<AuthenticatedUserDropdown />);
      expect(wrapper.snapshot).toMatchSnapshot();
    });
    test('without enterprise dashboard and expanded', () => {
      reduxHooks.useEnterpriseDashboardData.mockReturnValueOnce(null);
      useIsCollapsed.mockReturnValueOnce(false);
      const wrapper = shallow(<AuthenticatedUserDropdown />);
      expect(wrapper.snapshot).toMatchSnapshot();
    });
  });
});
