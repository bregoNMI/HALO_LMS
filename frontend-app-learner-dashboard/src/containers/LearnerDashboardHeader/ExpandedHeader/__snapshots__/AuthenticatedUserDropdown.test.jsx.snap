// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticatedUserDropdown snapshots no auth render empty 1`] = `null`;

exports[`AuthenticatedUserDropdown snapshots with enterprise dashboard 1`] = `
<Dropdown
  className="user-dropdown pr4"
>
  <Dropdown.Toggle
    className="p-4"
    id="user"
    src="profileImage"
    variant="light"
  >
    <span
      className="d-md-inline"
      data-hj-suppress={true}
    >
      username
    </span>
  </Dropdown.Toggle>
  <Dropdown.Menu
    className="dropdown-menu-right"
  >
    <Fragment>
      <Dropdown.Header>
        SWITCH DASHBOARD
      </Dropdown.Header>
      <Dropdown.Item
        as="a"
        className="active"
        href="/edx-dashboard"
      >
        Personal
      </Dropdown.Item>
      <Dropdown.Item
        as="a"
        href="url"
        key="label"
      >
        label
         
        Dashboard
      </Dropdown.Item>
      <Dropdown.Divider />
    </Fragment>
    <Dropdown.Item
      href="http://account-profile-url.test/u/username"
    >
      Profile
    </Dropdown.Item>
    <Dropdown.Item
      href="http://account-settings-url.test"
    >
      Account
    </Dropdown.Item>
    <Dropdown.Item
      href="http://order-history-url.test"
    >
      Order History
    </Dropdown.Item>
    <Dropdown.Divider />
    <Dropdown.Item
      href="http://logout-url.test"
    >
      Sign Out
    </Dropdown.Item>
  </Dropdown.Menu>
</Dropdown>
`;

exports[`AuthenticatedUserDropdown snapshots without enterprise dashboard and expanded 1`] = `
<Dropdown
  className="user-dropdown pr4"
>
  <Dropdown.Toggle
    className="p-4"
    id="user"
    src="profileImage"
    variant="light"
  >
    <span
      className="d-md-inline"
      data-hj-suppress={true}
    >
      username
    </span>
  </Dropdown.Toggle>
  <Dropdown.Menu
    className="dropdown-menu-right"
  >
    <Fragment>
      <Dropdown.Header>
        SWITCH DASHBOARD
      </Dropdown.Header>
      <Dropdown.Item
        as="a"
        className="active"
        href="/edx-dashboard"
      >
        Personal
      </Dropdown.Item>
      <Dropdown.Divider />
    </Fragment>
    <Dropdown.Item
      href="http://localhost:18000/career"
    >
      Career
      <Badge
        className="px-2 mx-2"
        variant="warning"
      >
        New
      </Badge>
    </Dropdown.Item>
    <Dropdown.Item
      href="http://account-profile-url.test/u/username"
    >
      Profile
    </Dropdown.Item>
    <Dropdown.Item
      href="http://account-settings-url.test"
    >
      Account
    </Dropdown.Item>
    <Dropdown.Item
      href="http://order-history-url.test"
    >
      Order History
    </Dropdown.Item>
    <Dropdown.Divider />
    <Dropdown.Item
      href="http://logout-url.test"
    >
      Sign Out
    </Dropdown.Item>
  </Dropdown.Menu>
</Dropdown>
`;
